pkg/oci/oci.go

```diff
@@ -73,15 +77,38 @@ func getLocalOciStore() (*oci.Store, error) {
 	return oci.New(defaultOciStore)
 }
 
-// GetGadgetImage pulls the gadget image and returns the a structure representing it.
-func GetGadgetImage(ctx context.Context, image string, authOpts *AuthOptions) (*GadgetImage, error) {
+// GetGadgetImage pulls the gadget image according to the pull policy and returns
+// a GadgetImage structure representing it.
+func GetGadgetImage(ctx context.Context, image string, authOpts *AuthOptions, pullPolicy string) (*GadgetImage, error) {
 	imageStore, err := getLocalOciStore()
 	if err != nil {
 		return nil, fmt.Errorf("getting local oci store: %w", err)
 	}
 
-	if err := pullIfNotExist(ctx, imageStore, authOpts, image); err != nil {
-		return nil, fmt.Errorf("pulling image %q: %w", image, err)
+	switch pullPolicy {
+	case PullImageAlways:
+		_, err := PullGadgetImage(ctx, image, authOpts)
+		if err != nil {
+			return nil, fmt.Errorf("pulling image %q: %w", image, err)
+		}
+		// We need to get the imageStore again because the previous call might have changed it
+		imageStore, err = getLocalOciStore()
+		if err != nil {
+			return nil, fmt.Errorf("getting local oci store: %w", err)
+		}
```

Comment:
What about splitting `PullGadgetImage` into two functions, one of them `pullGadgetImage` receiving the OCI store, then here we can call it passing `imageStore` and avoid this logic.

Reply:
Yes, that is the better way to resolve that